src/InterventionBackend.php

```diff
@@ -294,27 +324,102 @@ public function getQuality()
     }
 
     /**
-     * @return int The width of the image
+     * Return dimensions as array with cache enabled
+     *
+     * @return array Two-length array with width and height
      */
-    public function getWidth()
+    protected function getDimensions()
     {
+        // Default result
+        $result = [0, 0];
+
+        // We can only cache dimensions from the source file (before any manipulations are performed),
+        // so ensure that the original image is un-loaded into memory.
+        // If a manipulation has been performed, we will return the dimensions from the already loaded
+        // image resource. In any case the 'variant' for these manipulations is unknown, so we couldn't
+        // cache those anyway.
+        $resource = $this->image;
+        if ($resource) {
+            return [
```

Comment:
I'd use `getSize()` here, which returns a `Intervention\Image\Size` object with `->width` and `->height` public params. It looks like the underlying code will be called twice otherwise.

Note that `getSize()` is a magic method implemented by one of the GetSizeCommand classes in intervention.